### DISCUSSION TITLE
Introducing my blog

### DISCUSSION THREAD

**Discussion by McDic** [squidfunk/mkdocs-material#6631] [Show and tell]
Hello people,

I migrated my blog to `mkdocs-material` from [Jekyll TeXt Theme](https://github.com/kitian616/jekyll-TeXt-theme) due to several restrictions.

- Blog Main Page: https://blog.mcdic.net/
- Blog Architecture: https://blog.mcdic.net/architecture/

Thanks!

**Comment by McDic** [comment]
8 months after migrated the blog, my blog got some nice features on top of this mkdocs theme, all implemented myself.

### Auto-scrolling button

If you press the button at the top, you will be able to move to top/bottom of the page without putting additional effort.

![image](https://github.com/user-attachments/assets/113419ba-74ae-4817-88c3-846461514b17)

### Sorted posts

My blog sorts the posts by updated date, unique views, or created date.

- Recent posts: https://blog.mcdic.net/sorted/recent/
- Most viewed posts: https://blog.mcdic.net/sorted/most_viewed/
- Archives: https://blog.mcdic.net/archives/2021/ , etc
- Series: https://blog.mcdic.net/series/rust/ , etc

### Post statistics and history

For every post, statistics from Google analytics and history from git system are provided.

![image](https://github.com/user-attachments/assets/54c1c9c1-da28-45af-8519-c970239bbf3a)

**Comment by McDic** [comment]
### Random music player

Recently I added random music player from the Soundcloud. Users can optionally play random music from my playlist.

![image](https://github.com/user-attachments/assets/978cc92c-d1bb-474b-9caf-68d0e0ebcce8)

**Reply by McDic** [reply]
@Wcowin I overrided HTML templates and wrote some Javascripts to embed [Soundcloud Widget API](https://developers.soundcloud.com/docs/api/html5-widget), please have a look on [partials/headers/musicplayer.html](https://github.com/McDic/BlogV2/blob/master/overrides/partials/headers/musicplayer.html) and [javascripts/musicplayer.js](https://github.com/McDic/BlogV2/blob/master/docs/javascripts/musicplayer.js).

**Comment by LandonTClipp** [comment]
This is actually really cool. I was wondering how to best implement the view counter, I'm glad to see you made it work. I'm gonna have to borrow this :D

**Comment by McDic** [comment]
### Quiz pages

I introduced brain teaser quiz pages in my blog. Currently there are few, but I plan to make more. The quiz pages looks not very different from end user view, but it has some different internal mechanics and semantics when my custom blog plugin works for it.

[![image](https://github.com/user-attachments/assets/84f2c0ae-9b2f-4309-8b8a-fee6694cbfe4)](https://blog.mcdic.net/quiz/q1/question/)

### Random redirections

I've added random redirection pages for both my blog posts and quizzes, because now I think there are enough amount of personal blog posts to make random routing. Click [here](https://blog.mcdic.net/random/blog.html) to test random blog post.
